src/main/java/org/apache/sysml/api/python/SystemML.py

```diff
@@ -73,6 +391,42 @@ def __init__(self, sc, *args):
             self.sqlCtx = SQLContext(sc)
         except Py4JError:
             traceback.print_exc()
+            
+    def start(self):
+        try:
+            prevGlobalMLContext = MLContext.globalMLContext
+        except AttributeError:
+            pass
+        self.numTabs = 0
+        self.sqlCtx = SQLContext(self.sc)
+        MLContext.globalMLContext = self
+        self.reset()
+        self.pydml = []
+        self.matrixOutput = []
+        
+    
+    def eval(self, outputs=None):
+        if isinstance(outputs, Matrix) or isinstance(outputs, Scalar):
+            outputs.registerAsOutput()
+        elif not outputs is None:
+            for i in range(0, len(outputs)):
+                if isinstance(outputs[i], Matrix) or isinstance(outputs[i], Scalar):
+                    outputs[i].registerAsOutput()
+        if not outputs is None:
+            for o in self.matrixOutput:
+                self.registerOutput(o.getID())
+                addPyDMLStmt('save', args=[o.getID(), ', \" \"'])
+        pydml = ''.join(self.pydml)
+        # print('Execution following pydml script:\n' + pydml)
+        out = self.executeScript(pydml, isPyDML=True)
+        if not outputs is None:
```

Comment:
This check isn't necessary -- just start with the `for o in self.matrixOutput:` part, which will be fine even if `self.matrixOutput` is empty.